google/cloud/pubsub/internal/streaming_subscription_batch_source_test.cc

```diff
@@ -579,112 +562,159 @@ TEST(StreamingSubscriptionBatchSourceTest, AckMany) {
   EXPECT_THAT(done.get(), IsOk());
 }
 
-TEST(StreamingSubscriptionBatchSourceTest, ReadErrorWaitsForWrite) {
-  AutomaticallyCreatedBackgroundThreads background;
-  auto mock = std::make_shared<pubsub_testing::MockSubscriberStub>();
+CompletionQueue MakeMockCompletionQueue(AsyncSequencer<bool>& aseq) {
+  auto mock_cq = std::make_shared<MockCompletionQueueImpl>();
+  EXPECT_CALL(*mock_cq, MakeRelativeTimer)
+      .WillRepeatedly([&](std::chrono::nanoseconds) {
+        return aseq.PushBack("MakeRelativeTimer").then([](auto) {
+          return make_status_or(std::chrono::system_clock::now());
+        });
+      });
+  EXPECT_CALL(*mock_cq, RunAsync)
+      .WillRepeatedly([&](std::unique_ptr<RunAsyncBase> f) {
+        aseq.PushBack("RunAsync").then([function = std::move(f)](auto) mutable {
```

Comment:
Woo hoo!

Reply:
I can only imagine the glee with which @coryan typed that line